### DISCUSSION TITLE
An exhaustive word list for ispell and company-ispell.

### DISCUSSION THREAD

**Discussion by hongyi-zhao** [company-mode/company-mode#1156] [Q&A] [answered]
Currently, I'm using the following setting:

`company-ispell-dictionary (file-truename "~/Public/repo/github.com/ilius/Webster_s_Unabridged_3.txt")`

But I want to know how to let `company-ispell-dictionary` use multiple dictionaries stored in different files. 

Regards,
HY

**Comment by yugaego** [comment]
Hi HY,
It seems this question was [already discussed](https://github.com/company-mode/company-mode/issues/1146). Wasn't it resolved for you?

**Reply by hongyi-zhao** [reply]
No, the previous [issue](https://github.com/company-mode/company-mode/issues/1146) doesn't solve the problem of multiple dictionaries stored in different files. In fact, the solution I posted [there](https://github.com/company-mode/company-mode/issues/1146#issuecomment-886172208) only gives a method to create a customized dictionary file and use it as the value of `company-ispell-dictionary`, which is the default behavior supported by `company-ispell`.

**Reply by yugaego** [reply]
> create a customized dictionary file 

Yes, either this or my guess is `ispell-change-dictionary-hook` (available since Emacs 27.1) could be used to change `company-ispell-dictionary` value. These are two things I come up with (without adding changes to `company-ispell`).

**Reply by dgutov** [reply]
Any kind of different solution would have to start with settings for the built-in `ispell` package.

Can you make it work with multiple dictionaries the way you want? @hongyi-zhao If not, `M-x report-emacs-bug` is your friend.

I see some notion of "multiple dictionaries" there, but only one is used at a time, IIUC.

**Reply by hongyi-zhao** [reply]
I give some additional comments on using multiple dictionaries, even they can be used at the same time: 

1. All words in these dictionaries should not overlap.
2. All these dictionaries should be pre-ordered in lexicographic order to afford an acceptable autocompletion performance.

**Reply by dgutov** [reply]
Please see my comment above.

**Reply by hongyi-zhao** [reply]
I see what you mean: it has nothing to do with `company-ispell`; in essence, it should be supported by and  set for the built-in `ispell` package.

**Reply by hongyi-zhao** [reply]
See [here](https://lists.gnu.org/archive/html/help-gnu-emacs/2021-08/msg00138.html) for the relevant discussion.

**Reply by yugaego** [reply]
Hi @hongyi-zhao , I've skimmed through the discussion on the mailing list. It seems you had [posted the solution](https://github.com/hongyi-zhao/english-wordlist). If that is correct, feel free to write a new answer here and mark it as an accepted one.

[Probably the thread would have to be moved first from `General` to `Q&A` category: see the top of the sidebar.]

**Reply by hongyi-zhao** [reply]
> mark it as an accepted one.

Done. See the bottom reply, and I've marked it as an answer.
 
> [Probably the thread would have to be moved first from `General` to `Q&A` category: see the top of the sidebar.]

Done.

**Reply by yugaego** [reply]
> I don't know how to do this.

Yeah, we're all new to the Discussions, so it's a bit confusing.
It is [said in the Discussions docs](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#marking-a-comment-as-an-answer) to click a (circle-like) marker "in the upper-right corner of the" **parent** [added by me] "comment". 
[That's why I suggested to post a new (parent) answer/comment earlier.]
[And it seems to be quite a terminology mess with not that clear answer/comment distinction.]

**Reply by dgutov** [reply]
@hongyi-zhao You need to write a separate answer first, with the solution (so that people don't have to hunt for it inside the thread). Then we'll mark it as such.

**Comment by hongyi-zhao** [comment, marked as answer]
I created a new [project](https://github.com/hongyi-zhao/english-wordlist) based on [the discussion on the help-gnu-emacs](https://lists.gnu.org/archive/html/help-gnu-emacs/2021-08/msg00138.html). Please refer to it for a more detailed description. The following is the current number of entries contained in the word list file of this project:

```shell
$ wc -l american-english-exhaustive
875968 american-english-exhaustive
```
